TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Drivebase.java

```diff
@@ -0,0 +1,275 @@
+        brdrive.setPower(backRight / clamp);
+    }
+
+
+    /**
+     * For use in auto. Drives forward a specified distance, using the encoders.
+     *
+     * @param inches    The amount of inches to drive forward, negative is backwards.
+     * @param power     How fast to drive. (0, 1].
+     * @param telemetry Pass telemetry if you want this method to log to telemetry.
+     * @see Drivebase#driveSideways
+     * @see Drivebase#turnAngle
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void driveForward(double inches, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        double target = inches * ENCODER_PER_INCH;
+
+        fldrive.setTargetPosition((int) target);
+        frdrive.setTargetPosition((int) target);
+        bldrive.setTargetPosition((int) target);
+        brdrive.setTargetPosition((int) target);
+
+        fldrive.setPower(power);
+        frdrive.setPower(power);
+        bldrive.setPower(power);
+        brdrive.setPower(power);
+
+        setMotorModes(DcMotor.RunMode.RUN_TO_POSITION);
+
+        waitForMotors(telemetry);
+
+        fldrive.setPower(0);
+        frdrive.setPower(0);
+        bldrive.setPower(0);
+        brdrive.setPower(0);
+    }
+
+    /**
+     * For use in auto. Strafes sideways a specified amount of inches, using encoders.
+     *
+     * @param inches    How many inches to strafe, negative is left.
+     * @param power     How fast to strafe. (0, 1].
+     * @param telemetry Pass this if you want the method to log to telemetry.
+     * @see Drivebase#driveForward
+     * @see Drivebase#turnAngle
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void driveSideways(double inches, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        double target = inches * ENCODER_PER_INCH;
+
+        fldrive.setTargetPosition((int) target);
+        frdrive.setTargetPosition((int) -target);
+        bldrive.setTargetPosition((int) -target);
+        brdrive.setTargetPosition((int) target);
+
+        fldrive.setPower(power);
+        frdrive.setPower(-power);
+        bldrive.setPower(-power);
+        brdrive.setPower(power);
+
+        setMotorModes(DcMotor.RunMode.RUN_TO_POSITION);
+
+        waitForMotors(telemetry);
+
+        fldrive.setPower(0);
+        frdrive.setPower(0);
+        bldrive.setPower(0);
+        brdrive.setPower(0);
+    }
+
+    /**
+     * For auto. Turns to a specified angle, without resetting the heading.
+     *
+     * @param angle     How many degrees to turn. [-180, 180].
+     * @param power     How fast to turn. (0, 1].
+     * @param telemetry Pass this if you want this method to log to telemetry.
+     * @see Drivebase#driveSideways
+     * @see Drivebase#driveForward
+     * @see Drivebase#turnAngle
+     */
+    public @Api void turnToAngle(double angle, double power, @Nullable Telemetry telemetry) {
```

Comment:
Please reference https://github.com/Pleasant-Valley-Robotics/23-24_CenterStage6093/blob/3f340e439c2ae3dc7861254231e96d430fe44ada/FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/external/samples/RobotAutoDriveByGyro_Linear.java in order to improve your turning logic. This example uses a P controller to slow down as you approach the target and then automatically adjusts the heading. You can also reference this auto to improve your driveForward logic and your driveSideways logic to maintain a set heading, which will help account for motor drift if your robot is imbalanced.